base/docs/helpdb/Base.jl

```diff
@@ -2317,3 +2317,57 @@ for bit in (8, 16, 32, 64, 128)
         $(Symbol("UInt", bit))
     end
 end
+
+"""
+    Vector{T}(len)
+
+Construct an uninitialized [`Vector{T}`](@ref) of length `len`.
+
+# Examples
+```julia-repl
+julia> Vector{Float64}(3)
+3-element Array{Float64,1}:
+ 6.90966e-310
+ 6.90966e-310
+ 6.90966e-310
+```
+"""
+Vector{T}(len)
+
+"""
+    Matrix{T}(m, n)
+
+Construct an uninitialized [`Matrix{T}`](@ref) of size `m`×`n`.
+
+# Examples
+```julia-repl
+julia> Matrix{Float64}(2, 3)
+2×3 Array{Float64,2}:
+ 6.93517e-310  6.93517e-310  6.93517e-310
+ 6.93517e-310  6.93517e-310  1.29396e-320
+```
+"""
+Matrix{T}(m, n)
+
+"""
+    Array{T}(dims)
+    Array{T,N}(dims)
+
+Construct an uninitialized [`Array{T,N}`](@ref).
+`N` is determined from the length or number of `dims`. `dims` may
+be a tuple or a series of integer arguments corresponding to the lengths in each dimension.
+If the rank `N` is supplied explicitly as in `Array{T,N}(dims)`, then it must
+match the length or number of `dims`.
+
+# Examples
+```jldoctest
+julia> A = Array{Float64,2}(2, 2);
```

Comment:
> It would render as two blocks which is ugly

Yes, perhaps two  code-blocks that directly follow each other can be grouped into on? Note that in the REPL it does not show as two different blocks:
![screenshot from 2017-07-07 13-52-56](https://user-images.githubusercontent.com/11698744/27956874-03992d64-631c-11e7-9258-da33d20141e4.png)
![screenshot from 2017-07-07 13-53-15](https://user-images.githubusercontent.com/11698744/27956873-0398dba2-631c-11e7-83b8-aff28af39263.png)